ConcurrentModificationException occurred

stacktrace:

Caused by: java.util.ConcurrentModificationException
        at java.util.HashMap$HashIterator.nextNode(HashMap.java:1442)
        at java.util.HashMap$EntryIterator.next(HashMap.java:1476)
        at java.util.HashMap$EntryIterator.next(HashMap.java:1474)
        at com.microsoft.azure.maven.telemetry.AppInsightsProxy.mergeProperties(AppInsightsProxy.java:90)
        at com.microsoft.azure.maven.telemetry.AppInsightsProxy.trackEvent(AppInsightsProxy.java:68)
        at com.microsoft.azure.maven.telemetry.AppInsightsProxy.trackEvent(AppInsightsProxy.java:59)
        at com.microsoft.azure.maven.AbstractAzureMojo.trackMojoFailure(AbstractAzureMojo.java:332)
        at com.microsoft.azure.maven.AbstractAzureMojo.handleException(AbstractAzureMojo.java:344)
        at com.microsoft.azure.maven.AbstractAzureMojo.execute(AbstractAzureMojo.java:293)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
        ... 21 more

--- a/azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/telemetry/AppInsightsProxy.java
+++ b/azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/telemetry/AppInsightsProxy.java
@@ -12,7 +12,6 @@ import org.codehaus.plexus.util.StringUtils;
 import java.util.HashMap;
 import java.util.Iterator;
 import java.util.Map;
-import java.util.concurrent.ConcurrentHashMap;
 
 public class AppInsightsProxy implements TelemetryProxy {
     protected TelemetryClient client;

--- a/azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/telemetry/AppInsightsProxy.java
+++ b/azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/telemetry/AppInsightsProxy.java
@@ -10,7 +10,9 @@ import com.microsoft.applicationinsights.TelemetryClient;
 import org.codehaus.plexus.util.StringUtils;
 
 import java.util.HashMap;
+import java.util.Iterator;
 import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
 
 public class AppInsightsProxy implements TelemetryProxy {
     protected TelemetryClient client;
@@ -87,9 +89,10 @@ public class AppInsightsProxy implements TelemetryProxy {
             merged.putAll(customProperties);
             merged.putAll(defaultProperties);
         }
-        for (Map.Entry<String, String> entry : merged.entrySet()) {
-            if (StringUtils.isEmpty(entry.getValue())) {
-                merged.remove(entry.getKey());
+        Iterator<Map.Entry<String, String>> it = merged.entrySet().iterator();
+        while (it.hasNext()) {
+            if (StringUtils.isEmpty(it.next().getValue())) {
+                it.remove();
             }
         }
         return merged;

